### PR TITLE
Issue #9582: update example of AST for TokenTypes.TEXT_BLOCK_LITERAL_BEGIN

### LINKED ISSUE #9582: update example of AST for TokenTypes.TEXT_BLOCK_LITERAL_BEGIN
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#TEXT_BLOCK_LITERAL_BEGIN

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
fixes #9582 
BLOCK_LITERAL_BEGIN

Source code to generate code:
<pre>
<code>
public class test {
    public static void main(String[] args) {
        String hello = """
                 Hello, world!
                 """;
    }
}
</code>
</pre>

Output in AST Format:
<pre>
<code>
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--METHOD_DEF -> METHOD_DEF [2:4]
    |   |--MODIFIERS -> MODIFIERS [2:4]
    |   |   |--LITERAL_PUBLIC -> public [2:4]
    |   |   `--LITERAL_STATIC -> static [2:11]
    |   |--TYPE -> TYPE [2:18]
    |   |   `--LITERAL_VOID -> void [2:18]
    |   |--IDENT -> main [2:23]
    |   |--LPAREN -> ( [2:27]
    |   |--PARAMETERS -> PARAMETERS [2:34]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [2:34]
    |   |       |--MODIFIERS -> MODIFIERS [2:34]
    |   |       |--TYPE -> TYPE [2:34]
    |   |       |   `--ARRAY_DECLARATOR -> [ [2:34]
    |   |       |       |--IDENT -> String [2:28]
    |   |       |       `--RBRACK -> ] [2:35]
    |   |       `--IDENT -> args [2:37]
    |   |--RPAREN -> ) [2:41]
    |   `--SLIST -> { [2:43]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [3:8]
    |       |   |--MODIFIERS -> MODIFIERS [3:8]
    |       |   |--TYPE -> TYPE [3:8]
    |       |   |   `--IDENT -> String [3:8]
    |       |   |--IDENT -> hello [3:15]
    |       |   `--ASSIGN -> = [3:21]
    |       |       `--EXPR -> EXPR [3:23]
    |       |           `--TEXT_BLOCK_LITERAL_BEGIN -> """ [3:23]
    |       |               |--TEXT_BLOCK_CONTENT -> \r\n                 Hello, world!\r\n                  [3:26]
    |       |               `--TEXT_BLOCK_LITERAL_END -> """ [5:17]
    |       |--SEMI -> ; [5:20]
    |       `--RCURLY -> } [6:4]
    `--RCURLY -> } [7:0]

</code>
</pre>

Expected Java Doc to be Updated:
<pre>
<code>
|--VARIABLE_DEF -> VARIABLE_DEF
 |   |--MODIFIERS -> MODIFIERS
 |   |--TYPE -> TYPE
 |   |   `--IDENT -> String
 |   |--IDENT -> hello
 |   `--ASSIGN -> =
 |       `--EXPR -> EXPR
 |           `--TEXT_BLOCK_LITERAL_BEGIN -> """
 |               |--TEXT_BLOCK_CONTENT -> \n                Hello, world!\n
 |               `--TEXT_BLOCK_LITERAL_END -> """
 |--SEMI -> ;
</code>
</pre>

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
@pbludov  please review my changes!

**Comment by Rahulkhinchi03** [comment]
@AkMo3 why the wercker/build is still pending from yesterday?

**Comment by Rahulkhinchi03** [comment]
@timurt and @pbludov  please review changes and approve the same!

**Review by pbludov — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -3875,16 +3875,16 @@
...
-     *     |
-     *     +--IDENT (ArrayList)
-     *     +--LPAREN (()
-     *     +--ELIST
-     *         |
-     *         +--EXPR
-     *             |
-     *             +--NUM_INT (50)
-     *     +--RPAREN ())
+     * LITERAL_NEW -&gt; new
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3878]: why this changes are here? they are related to `LITERAL_NEW`

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -5274,10 +5274,10 @@
      * |   |   `--IDENT -&gt; String
      * |   |--IDENT -&gt; hello
      * |   `--ASSIGN -&gt; =
-     * |       `--EXPR -&gt; EXPR
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:5277]: it seems idents were correct, please double check with ast fragment you provided

```
 |   `--ASSIGN -> =
 |       `--EXPR -> EXPR
 |           `--TEXT_BLOCK_LITERAL_BEGIN -> """
 |               |--TEXT_BLOCK_CONTENT -> \n                Hello, world!\n
 |               `--TEXT_BLOCK_LITERAL_END -> """
 |--SEMI -> ;
```

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3878]: I did pull request of LITERAL_NEW yesterday and I don't know why it is showing here too.  I am trying to resolve this.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:5277]: The idents were correct but spacing was not correct and also the previous person committed it and he closed it and maybe it made chaos. Still, can you help me to resolve this @timurt ?

**Comment by Rahulkhinchi03** [comment]
I am closing this pull request as I lost my branch.
